src/main/java/com/arkvis/orchid/PeriodCalendar.java

```diff
@@ -21,4 +24,44 @@ public void addMenstruation(LocalDate date) {
         day.addMenstruation();
         dayMap.put(date, day);
     }
+
+    public LocalDate getNextMenstruationDate() {
+        List<Day> cycleStartDays = getCycleStartDays();
+        if (cycleStartDays.isEmpty()) return null;
+
+        Day lastCycleStartDay = getLastItem(cycleStartDays);
+        long averageCycleLength = getAverageCycleLength(cycleStartDays);
+        return lastCycleStartDay.getDate().plusDays(averageCycleLength);
+    }
+
+    private long getAverageCycleLength(List<Day> cycleStartDays) {
+        if (cycleStartDays.size() <= 1) return DEFAULT_CYCLE_LENGTH_IN_DAYS;
+
+        LocalDate startDate = cycleStartDays.get(0).getDate();
+        LocalDate endDate = getLastItem(cycleStartDays).getDate();
+
+        long totalDays = DAYS.between(startDate, endDate) + DEFAULT_CYCLE_LENGTH_IN_DAYS;
+        return totalDays / cycleStartDays.size();
+    }
+
+    private List<Day> getCycleStartDays() {
```

Comment:
Can we create an object like "PeriodPredictor" to encapsulate the prediction functionality?

Reply:
Done